server/map_rpc_server.go

```diff
@@ -697,14 +674,36 @@ func (t *TrillianMapServer) snapshotForTree(ctx context.Context, tree *trillian.
 	return tx, err
 }
 
-// hasDuplicates returns an error if there are duplicates in indexes.
-func hasDuplicates(indexes [][]byte) error {
-	set := make(map[string]bool)
-	for _, i := range indexes {
-		if set[string(i)] {
-			return status.Errorf(codes.InvalidArgument, "index %x requested more than once", i)
-		}
-		set[string(i)] = true
+// requestIndexValidator validates the indices in a single map request.
+type requestIndexValidator struct {
```

Comment:
Takes a while to guess how the nouns are related in this name. How about `leafIndexValidator`?

Reply:
No longer exists.